tests/vmi_multus_test.go

```diff
@@ -927,22 +928,21 @@ var _ = Describe("[Serial]Macvtap", func() {
 		tests.DisableFeatureGate(virtconfig.MacvtapGate)
 	})
 
-	newRandomVMIWithMacvtapInterfaceEphemeralDiskAndUserdata := func(containerImage string, userData string, macvtapNetworkName string) *v1.VirtualMachineInstance {
-		vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(containerImage, userData)
-		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMacvtapNetworkInterface(macvtapNetworkName)}
-		vmi.Spec.Networks = []v1.Network{{
-			Name: macvtapNetworkName,
-			NetworkSource: v1.NetworkSource{
-				Multus: &v1.MultusNetwork{
-					NetworkName: macvtapNetworkName,
-				},
-			},
-		}}
+	newCirrosVMIWithMacvtapNetwork := func(macvtapNetworkName string) *v1.VirtualMachineInstance {
+		macvtapMultusNetwork := libvmi.MultusNetwork(macvtapNetworkName)
+		vmi := libvmi.NewCirros(
+			libvmi.WithInterface(*v1.DefaultMacvtapNetworkInterface(macvtapNetworkName)),
+			libvmi.WithNetwork(&macvtapMultusNetwork))
+
+		// without this, the VMI instantiation fails due to the following k8s
+		// API error:
+		// "an empty namespace may not be set when a resource name is provided"
+		vmi.SetNamespace(tests.NamespaceTestDefault)
```

Comment:
You use the same method on [services.go](https://github.com/kubevirt/kubevirt/blob/b5e725fec936c26b2dc5406b7e9574fa6c7d51b0/tests/network/services.go#L132) tests, how does it work there?

Reply:
Services tests do not rely on Multus - macvtap does. 

KubeVirt sets annotations on the pod requesting a resource for each multus network. That thing, is prevented by kubernetes.

Reply:
What do you mean by "the pod does not feature the namespace in its specification"? Sounds like a bug in `ibvmi.vmi.baseVmi` it should create vmi's in the NamespaceTestDefault and not passing an empty namespace. Isn't it?
@EdDev

Reply:
The namespace should not be set by the library (I removed that). On creation you specify the namespace (`kubecli.VirtualMachineInstance("namespace").Create()`. this vmi library had the wrong assumption that the namespace in the metadata is responsible for creating the VMI in the desired namespace.

The k8s cluster fills in the namespace in the  namespace on the  creation (along with other metadata). The pattern to go is using the returned VMI from the client call:

```
vmi, err = kubecli.VirtualMachine("namespace").Create(vmi)
```

The returned VMI has everything correctly set.

Taking the usage of this funnction into account, one example usage could look like this:

```
vmi = tests.StartVMIOnNode(vmi, nodeName)
Expect(vmi.Namespace).ToNot(BeEmpty())
```

Reply:
@rmohr thanks for the clarification.
@maiqueb please  remove the namespace setting and check if using the returned vmi on `vmi = tests.StartVMIOnNode(vmi, nodeName)` solves your issue.

Reply:
It did.

Done.